Framework/Algorithms/test/CalcCountRateTest.h

```diff
@@ -0,0 +1,478 @@
+    alg.setSearchRanges(sws);
+    TS_ASSERT_THROWS_NOTHING(alg.setVisWS("testVisWSNorm"));
+
+    auto visNormLog = alg.getVisNormLog();
+    TS_ASSERT_EQUALS(visNormLog.size(), 100);
+    TS_ASSERT_DELTA(visNormLog[10], 2, 1.e-5);
+    auto sum = std::accumulate(visNormLog.begin(), visNormLog.end(), 0.);
+
+    TS_ASSERT_DELTA(sum, 200., 1.e-4);
+  }
+
+  void test_visWS_NormalizationCoarce() {
+
+    auto sws = build_test_ws(true);
+
+    CalcCountRateTester alg;
+    alg.initialize();
+
+    alg.setProperty("NumTimeSteps", 50);
+    alg.setProperty("XResolution", 200);
+
+    alg.setProperty("RangeUnits", "dSpacing");
+
+    alg.setProperty("NormalizeTheRate", true);
+    alg.setProperty("UseLogDerivative", true);
+    alg.setProperty("UseNormLogGranularity", true);
+
+    alg.setProperty("Workspace", sws);
+
+    alg.setOutLogParameters(sws);
+    alg.setSearchRanges(sws);
+    TS_ASSERT_THROWS_NOTHING(alg.setVisWS("testVisWSNorm"));
+
+    auto visNormLog = alg.getVisNormLog();
+    TS_ASSERT_EQUALS(visNormLog.size(), 50);
+    TS_ASSERT_DELTA(visNormLog[10], 4, 1.e-5);
+    auto sum = std::accumulate(visNormLog.begin(), visNormLog.end(), 0.);
+
+    TS_ASSERT_DELTA(sum, 200., 1.e-4);
+  }
+
+  void test_visWS_Normalized() {
+
+    auto sws = build_test_ws(true);
+
+    CalcCountRateTester alg;
+    alg.initialize();
+
+    alg.setProperty("NumTimeSteps", 50);
+    alg.setProperty("XResolution", 200);
+    alg.setProperty("VisualizationWs", "testVisWSNormalized");
+
+    alg.setProperty("RangeUnits", "dSpacing");
+
+    alg.setProperty("NormalizeTheRate", true);
+    alg.setProperty("UseLogDerivative", true);
+    alg.setProperty("UseNormLogGranularity", true);
+
+    alg.setProperty("Workspace", sws);
+
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+
+    API::MatrixWorkspace_sptr testVisWS = alg.getProperty("VisualizationWs");
+    TS_ASSERT(testVisWS);
+    if (!testVisWS)
+      return;
+    TS_ASSERT_EQUALS(testVisWS->getNumberHistograms(), 50);
+    const MantidVec &X = testVisWS->readX(0);
+    const MantidVec &Y = testVisWS->readY(0);
+    TS_ASSERT_EQUALS(X.size(), 201);
+    TS_ASSERT_EQUALS(Y.size(), 200);
+    auto Yax = dynamic_cast<API::NumericAxis *>(testVisWS->getAxis(1));
+    TS_ASSERT(Yax);
+    if (!Yax)
+      return;
+
+    // auto &YaxVal = Yax->getValues();
+
+    auto newLog = dynamic_cast<Kernel::TimeSeriesProperty<double> *>(
+        sws->run().getLogData("block_count_rate"));
+    TS_ASSERT(newLog);
+    if (!newLog)
+      return;
+    // these times are shifted by start time so firts time is 0
+    // auto times = newLog->timesAsVectorSeconds();
+    MantidVec counts = newLog->valuesAsVector();
+
+    for (size_t i = 0; i < testVisWS->getNumberHistograms(); ++i) {
+      const MantidVec &Y = testVisWS->readY(i);
+      double sum = std::accumulate(Y.begin(), Y.end(), 0.);
+      TSM_ASSERT_DELTA("Incorrect counts at index: " +
+                           boost::lexical_cast<std::string>(i),
```

Comment:
boost claims its the fastest, though it not important in this case. I also remember some problems with this on some OS (not Windows). But let's try.